src/erlfmt_format.erl

```diff
@@ -312,21 +310,23 @@ binary_op_to_algebra(Op, Meta, Left, Right, Indent) ->
     IsNextBreakFits = is_next_break_fits_op(Op, Right) andalso not HasBreak,
 
     Doc =
-        case Op of
-            '|' ->
-                 with_next_break_fits(IsNextBreakFits, RightD, fun (R) ->
-                    RightOpD = nest(break(concat(<<" ">>, OpD), R), Indent, break),
-                    concat(LeftD, concat(maybe_force_breaks(HasBreak), RightOpD))
-                end);
-            _ ->
-                with_next_break_fits(IsNextBreakFits, RightD, fun (R) ->
-                    RightOpD = nest(break(concat(<<" ">>, OpD), group(R)), Indent, break),
-                    concat(group(LeftD), group(concat(maybe_force_breaks(HasBreak), RightOpD)))
-                end)
-        end,
-
+        with_next_break_fits(IsNextBreakFits, RightD, fun (R) ->
+            RightOpD = nest(break(concat(<<" ">>, OpD), group(R)), Indent, break),
+            concat(group(LeftD), group(concat(maybe_force_breaks(HasBreak), RightOpD)))
+        end),
     combine_comments(Meta, maybe_wrap_in_parens(Meta, Doc)).
 
+union_to_algebra(Meta, Left, Right) ->
+    Unions = [Left | collect_unions(Right)],
```

Comment:
I guess we are relying a bit on how the parser disambiguates by not calling `collect_unions(Left)`

Reply:
Yeah, we rely on the fact that it's right-associative and recurse only on the right side. If there is a nested union on the left as well, it had to be wrapped in parens to even parse and probably should be treated as a separate expression.